test/src/kernels/EFieldAdvection.C

```diff
@@ -33,8 +33,9 @@ EFieldAdvection::EFieldAdvection(const InputParameters & parameters)
   : Kernel(parameters),
     _efield_id(coupled("efield")),
     _efield(coupledVectorValue("efield")),
-    _efield_var(*getVectorVar("efield", 0)),
-    _vector_phi(_assembly.phi(_efield_var)),
+    _efield_coupled(isCoupled("efield")),
+    _efield_var(_efield_coupled ? getVectorVar("efield", 0) : nullptr),
+    _vector_phi(_efield_coupled ? &_assembly.phi(*_efield_var) : nullptr),
```

Comment:
Another alternative is to reference a dummy object so you don't have to use a pointer in the member methods. Up to you.

Reply:
I kind of like the explicit nullness